### PR TITLE
Rework validation

### PR DESCRIPTION
**Pull request by piercus**
Rework the way the `validation_step` is done.

* Grid vizualisation 
* PIQ metrics aggregated over the whole validation

Outside scope : 
* Remove wandb and tensorboard loggers for now, cause current changes make them obsolete
* Multi-gpu metric aggregation (will do it as a follow-up step)
* Log training samples

### Legend

| | First Column | Second Column | Third Column |
|---|---|---|---|
| First Row | Before image | After Image (Ground truth) | Mask |
| Second Row | 1-steps-pred | 4-steps-pred | <blank> |
| Third Row | 1-steps-pred vs Before | 4-steps-pred vs before | <blank> |

### Example

a014a148-b95b-448a-ac8d-4b53f4099fcb

### CODE REVIEW

File: src/lbm/models/base/base_model.py [hunk 1]
```diff
@@ -49,10 +49,6 @@ def to(self, *args, **kwargs):
             self.dtype = dtype
         return self
 
-    def compute_metrics(self, batch: Dict[str, Any], *args, **kwargs):
```
Comment (deltheil) [inline src/lbm/models/base/base_model.py:52]: Why?

File: src/lbm/models/lbm/lbm_model.py [hunk 2]
```diff
@@ -501,7 +501,7 @@ def log_samples(
                 z = source_image
 
             with torch.autocast(dtype=self.dtype, device_type="cuda"):
-                logs[f"samples_{num_step}_steps"] = self.sample(
+                logs[f"{num_step}_steps"] = self.sample(
```
Comment (deltheil) [inline src/lbm/models/lbm/lbm_model.py:504]: Why?

File: requirements.txt [hunk 3]
```diff
@@ -22,4 +22,5 @@ transformers==4.42.3
 wandb==0.16.2
 webdataset>=0.2.86
 kornia==0.8.0
-neptune>=1.0
\ No newline at end of file
+neptune>=1.0
+piq>=0.8.0
```
Comment (deltheil) [inline requirements.txt:26]: Maybe we could directly use torchmetrics (already a dependency)? (PieApp is missing though)

File: src/lbm/trainer/training_config.py [hunk 4]
```diff
@@ -70,6 +70,7 @@ class TrainingConfig(BaseConfig):
     lr_scheduler_frequency: Optional[int] = 1
     metrics: Optional[List[str]] = None
     tracking_metrics: Optional[List[str]] = None
+    column_visualization_keys: List[Tuple[str, str]] = field(default_factory=lambda: [])
```
Comment (deltheil) [inline src/lbm/trainer/training_config.py:73]: You should add it to the above docstring. But also, naming could be improved (maybe).

File: src/lbm/trainer/loggers.py [hunk 5]
```diff
@@ -92,238 +92,6 @@ def wrap_text(
     lines.append(current_line)
     return lines
 
-
-class WandbSampleLogger(Callback):
```
Comment (deltheil) [inline src/lbm/trainer/loggers.py:96]: Well, it will break `examples/training/train_lbm_surface.py`. Can't we find a middle ground here?

**Review by deltheil — COMMENTED** [review]
Some nit-s and questions.

File: src/lbm/models/lbm/lbm_model.py (hunk 2 above)
Comment (piercus) [reply src/lbm/models/lbm/lbm_model.py:504]: In current version the metrics names are derived from this key "samples_{num_step}_steps" makes `pieapp_samples_{num_step}_steps` which i feel is too verbose compared to just `pieapp_{num_step}_steps`
